src/pip/_internal/cli/base_command.py

```diff
@@ -141,7 +141,7 @@ def _main(self, args):
             not options.no_python_version_warning
         ):
             message = (
-                "A future version of pip will drop support for Python 2.7. "
+                "pip 21.0 will drop support for Python 2.7. "
```

Comment:
Perhaps adding the date will make it more concrete and actionable:

```suggestion
                "pip 21.0 will drop support for Python 2.7 in January 2021. "
```